modules/nw-udn-cr.adoc

```diff
@@ -34,32 +34,89 @@ spec:
+====
+
+.. Create a YAML file, such as `my-layer-three-udn.yaml`, to define your request as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-2-primary # <1>
+  namespace: <some_custom_namespace>
+spec:
+  topology: Layer3 # <2>
+  layer3: # <3>
+    role: Primary # <4>
+    mtu: 1300 # <5>
+    subnets: # <6>
+      - cidr: 10.150.0.0/16
+        hostSubnet: 24
+      - cidr: 2001:db8::/60
+        hostSubnet: 64
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
+<2> The `topology` field describes the network configuration, accepted values are `Layer2` and `Layer3`. Specifying a `Layer3` topology type creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+<3> This field specifies the topology configuration, valid values are `layer2` or `layer3`.
+<4> Specifies a `Primary` or `Secondary` role. `Primary` is the only `role` specification supported in {product-version}.
+<5> The maximum transmission units (MTU) is an optional field. The default value is `1400`. The boundary for IPv4 is `574`, and for IPv6 it is `1280`.
+<6> For `Layer3` topology types the following specifies config details for the `subnet` field:
++
+* The `subnets` field is mandatory.
+* The type for the `subnets` field is `cidr` and `hostSubnet`:
+** `cidr` is the cluster subnet and accepts a string value.
+** `hostSubnet` specifies the nodes subnet prefix that the cluster subnet is split to.
+** For IPv6, only a `/64` length is supported for `hostSubnet`.
++
+[NOTE]
+====
+There is an additional configurations that UDN uses for a `Layer3` topology, `spec.joinSubnets`. It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+
+When omitted, the platform will set as a default value for IPv4 `100.65.0.0/16` and for IPv6 `fd99::/64`. If the default address values are used anywhere in the cluster's network you must override it by setting the `joinSubnets` field. If you choose to set this field, ensure it does not conflict with other subnets in the cluster such as the cluster subnet, the `default` network cluster subnet, and the masquerade subnet.
+
+The `joinSubnets` field configures the routing between different segments within the Layer 2 network. Dual-stack clusters may set 2 subnets, one for each IP family, otherwise only 1 subnet is allowed. This field is only allowed for the `Primary` network.
```

Comment:
If you use a semicolon above, consider using it here: 

```suggestion
The `joinSubnets` field configures the routing between different segments within the Layer 2 network. Dual-stack clusters might set 2 subnets, one for each IP family; otherwise only 1 subnet is allowed. This field is only allowed for the `Primary` network.
```